fed/mosaic-mapping/src/test/java/org/eclipse/mosaic/fed/mapping/ambassador/SpawningFrameworkTrafficLightTest.java

```diff
@@ -0,0 +1,140 @@
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests for {@link SpawningFramework}.
+ */
+public class SpawningFrameworkTrafficLightTest {
+
+    @Mock
+    public RtiAmbassador rti;
+
+    private RandomNumberGenerator rng;
+
+    @Before
+    public void setup() {
+        rng = new DefaultRandomNumberGenerator(0L);
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void initAppsWithoutWeight() throws InternalFederateException, IllegalValueException {
+        //SETUP
+        CMappingAmbassador mappingConfig = new CMappingAmbassador();
+        mappingConfig.trafficLights = new ArrayList();
+        mappingConfig.trafficLights.add(newTlConfig(null, null, "app_1"));
+        mappingConfig.trafficLights.add(newTlConfig( null, null, "app_2"));
+        ScenarioTrafficLightRegistration tlRegistration = newTlRegistration(1000);
+        ArgumentCaptor<TrafficLightRegistration> captor = ArgumentCaptor.forClass(TrafficLightRegistration.class);
+
+        //RUN
+        SpawningFramework spawningFramework = new SpawningFramework(mappingConfig, tlRegistration, rti, rng);
+        spawningFramework.timeAdvance(0, rti, rng);
+
+        //ASSERT
+        verify(rti, times(1000)).triggerInteraction(captor.capture());
+        List <TrafficLightRegistration> tlrs = captor.getAllValues();
+        Assert.assertEquals(countAppInTlr(tlrs, "app_1") / 1000d, 0.5d, 0.02d);
+        Assert.assertEquals(countAppInTlr(tlrs, "app_2") / 1000d, 0.5d, 0.02d);
+    }
+
+    @Test
+    public void initAppsWithWeightAndTlGroupId() throws InternalFederateException, IllegalValueException {
+        //SETUP
+        CMappingAmbassador mappingConfig = new CMappingAmbassador();
+        mappingConfig.trafficLights = new ArrayList();
+        mappingConfig.trafficLights.add(newTlConfig(null, 0.2, "app_1"));
+        mappingConfig.trafficLights.add(newTlConfig( null, 0.8, "app_2"));
+        mappingConfig.trafficLights.add(newTlConfig( "13", null, "app_3"));
+        ScenarioTrafficLightRegistration tlRegistration = newTlRegistration(1000);
+        ArgumentCaptor<TrafficLightRegistration> captor = ArgumentCaptor.forClass(TrafficLightRegistration.class);
+
+        //RUN
+        SpawningFramework spawningFramework = new SpawningFramework(mappingConfig, tlRegistration, rti, rng);
+        spawningFramework.timeAdvance(0, rti, rng);
+
+        //ASSERT
+        verify(rti, times(1000)).triggerInteraction(captor.capture());
+        List <TrafficLightRegistration> tlrs = captor.getAllValues();
+        Assert.assertEquals(countAppInTlr(tlrs, "app_1") / 1000d, 0.2d, 0.02d);
+        Assert.assertEquals(countAppInTlr(tlrs, "app_2") / 1000d, 0.8d, 0.02d);
+        Assert.assertEquals(tlrs.get(13).getMapping().getApplications().get(0), "app_3");
+    }
```

Comment:
I like that :) thanks! 
However, it would be great to see another test to see an equal distribution in the case `tlGroupId == null && weight == null`

Reply:
a sorry! I overlooked that it's already there 🙈